### DISCUSSION TITLE
ArchR is running safely in R CLI, but "Cairo is not functional" error shows in rstudio, even Cairo is already installed.

### DISCUSSION THREAD

**Discussion by genomicworks** [GreenleafLab/ArchR#1915] [Questions / Documentation] [answered]
I think I safely installed ArchR, and confimed that ArchR is running safely in R command line. 
But when I run the ArchR package in rstudio, 

"WARNING : Cairo check shows Cairo is not functional.          
 ggplot2 rasterization will not be available without Cario.
 This may cause issues editing plots with many thousands of points from single cells." error comes. 

But, Cairo package was already installed, (with library(Cairo) showed no error)
and the warning is not shown in R command line. 

I think this is the difference between rstudio and R CLI. but I can't solve it. 

In the ArchR code, 
if (!.checkCairo()){
    packageStartupMessage("WARNING : Cairo check shows Cairo is not functional.\n          ggplot2 rasterization will not be available without Cario.\n          This may cause issues editing plots with many thousands of points from single cells.")
  }

Why ArchR could not check the already installed Cairo package in rstudio?
Thank you very much, everybody, firstly.

**Comment by rcorces** [comment]
Sounds like your different R setups are seeing different environments. Cairo is both a system library and an R package and both are required for Cairo to be functional.

**Reply by genomicworks** [reply]
Thank you very much for your quick response, Sir. 

I checked .libPaths() in both environment, and results the same library paths are used. 
Only R4.0.2 is install in my Centos7 computer. 

If you can help me more, 
could you tell me some more detail about what point I can check?

Thank you firstly, sir.

**Reply by rcorces** [reply, marked as answer]
I do not have additional input on this. The `.checkCairo()` function is very simple. You will have to diagnose for yourself why it is failing.

https://github.com/GreenleafLab/ArchR/blob/f6c0388bd37023400794c9ae8562ad69e3ba9fd7/R/GgplotUtils.R#L956-L966

Also not that Cairo is not necessarily essential for ArchR.

**Reply by genomicworks** [reply]
Thank you very much, rcorces,

I'll try by myself. (not optimistic~^^)

**Comment by sherry-mm** [comment]
I have the same problem as you. Have you solved it? Thank you very much for your reply

**Comment by um-pdavila** [comment]
This may be a bug in ArchR package requirements.  Updating cario R package fixed issue for us.
